### PR TITLE
Add documentation on rehearsals

### PR DESCRIPTION
**Pull request by petr-muller**
/cc @stevekuznetsov @droslean @bbguimaraes

### CODE REVIEW

File: ci-operator/REHEARSALS.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+trigger a so-called "rehearsal". Jobs that would be affected by a PR are
+executed as if run against a target component repository after job changes
+would be merged. This provides job config authors early feedback about how job
+config changes impact CI setup for a given repo.
+
+## How rehearsal works
+
+All pull requests trigger a `ci/prow/pj-rehearse` check, which detects the jobs
+affected by the proposed change. For each job selected for rehearsal (for
+various reasons, we are not able to rehearse all jobs), new Prowjob is
```
Comment (droslean) [inline ci-operator/REHEARSALS.md]: You already mention this parenthesis below

File: ci-operator/REHEARSALS.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+1. Newly added presubmit jobs
+2. Presubmit jobs whose `spec:` field was changed
+
+These jobs are then further filtered to exclude jobs where some factor prevents
+us from reliably rehearsing them. Decisions about why a job was excluded from
+rehearsing are logged in the output of the `ci/prow/pj-rehearse` job. Most
+importantly, we are currently *not* rehearsing template-based jobs.
+
+## Future work
```
Comment (droslean) [inline ci-operator/REHEARSALS.md]: Worth to mention, that in the future we will rehearse job not only for changed `spec`, but for other fields as well.

File: ci-operator/REHEARSALS.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+CI job rehearsals
+=================
+
+**This feature is not fully deployed to production yet**
```
Comment (stevekuznetsov) [inline ci-operator/REHEARSALS.md]: This needs to be put into words a user will understand -- what is and is not deployed? What works? Maybe just specify what _is_ working

File: ci-operator/REHEARSALS.md (hunk 1 above)
Comment (petr-muller) [reply ci-operator/REHEARSALS.md]: That's intentional. Not everyone reads the whole thing. Here I mention it as a fact, and below I explain details about it.

File: ci-operator/REHEARSALS.md (hunk 2 above)
Comment (petr-muller) [reply ci-operator/REHEARSALS.md]: addressed

File: ci-operator/REHEARSALS.md (hunk 3 above)
Comment (petr-muller) [reply ci-operator/REHEARSALS.md]: addressed

**Comment by petr-muller** [comment]
I have addressed the feedback in a new commit so it's easier to see how it was changed. I'd like to squash those commits before merge.

/hold
/retest

File: ci-operator/REHEARSALS.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+affected by the proposed change. For each job selected for rehearsal (for
+various reasons, we are not able to rehearse all jobs), new Prowjob is
+dynamically created, reporting results via a GitHub context named with the
+`ci/rehearse/$org/$repo/$test` pattern.
+
+## Which jobs are rehearsed
+
+At the moment, we are not able to rehearse all affected jobs for various
+reasons. Currently, we only rehearse `Presubmit` jobs affected by the
+following changes:
```
Comment (stevekuznetsov) [inline ci-operator/REHEARSALS.md]: We must be diligent about updating this

**Pull request merged**